src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -76,6 +78,126 @@ def chunk_iter(inputs, feature_extractor, chunk_len, stride_left, stride_right):
+        sequence, stride = item
+        chunk_len, stride_left, stride_right = stride
+        sequence = sequence.squeeze(0)
+        sequence = sequence[begin_idx:]
+        if seq_idx != 0:
+            time += (chunk_len - stride_left) / 100
+            timestamp_tokens = np.where(sequence >= timestamp_begin)[0][0::2]
+            previous_tokens = items[-1][1:-1]
+
+            if len(timestamp_tokens) > 1 and len(previous_tokens) > 0:
+                current_slice = sequence[: timestamp_tokens[1]]
+
+                trie = SuffixTrie(current_slice)
+                longest_common_sequence = trie.search(previous_tokens)
+                if len(longest_common_sequence) == len(previous_tokens):
+                    idx = longest_common_sequence[0][0]
+                    sliced_sequence = sequence[idx : timestamp_tokens[1]]
+
+                    actual_offset = int(time / time_precision) + timestamp_begin
+                    sliced_sequence[-1] += actual_offset
+                    items[-1][1:] = sliced_sequence
+                    sequence = sequence[timestamp_tokens[1] :]
+
+        timestamp_tokens = sequence >= timestamp_begin
+        consecutive = np.where(timestamp_tokens[:-1] & timestamp_tokens[1:])[0] + 1
+        if len(consecutive) > 0:  # if the output contains two consecutive timestamp tokens
+            last_slice = 0
+            # take the last timestamp of the previous chunk
+            actual_offset = int(time / time_precision) + timestamp_begin
+            for current_slice in consecutive:
+                sliced_tokens = sequence[last_slice:current_slice]
+                # set correct timestamps
+                sliced_tokens[0] += actual_offset - timestamp_begin
+                sliced_tokens[-1] += actual_offset - timestamp_begin
+                items.append(sliced_tokens)  # correct final sequence
+                last_slice = current_slice
+            if np.where(timestamp_tokens)[0][-1] != current_slice:  # we probably have a timestamp at the end
+                # offset = items[-1][-1] if len(items) > 0 else timestamp_begin
+                sliced_tokens = sequence[current_slice : np.where(timestamp_tokens)[0][-1] + 1]
+                sliced_tokens[0] += actual_offset - timestamp_begin
+                sliced_tokens[-1] += actual_offset - timestamp_begin
+                items.append(sliced_tokens)  # correct final sequence
+        else:
+            timestamps = sequence[timestamp_tokens.nonzero()[0].flatten()]
+            if len(timestamps) > 0 and timestamps[-1].item() != timestamp_begin:
+                # no consecutive timestamps but it has a timestamp; use the last one.
+                # single timestamp at the end means no speech after the last timestamp.
+                items.append(sequence[np.where(timestamps[-1])])
+
+    result = []
+    for i in range(len(items)):
+        result += items[i].tolist()
+    return result
+
+
+def _fast_find_longest_common_sequence(sequences, tokenizer):
```

Comment:
I think this is correct, but suffers from issues with being not as robust to tiny token error. (like Sir vs sir in the stride).

I haven't retried my more contrived examples, but overall it seemed to perform less well on real data than the more naive and slow algorithm.